edgelet/iotedge/src/check/mod.rs

```diff
@@ -1460,6 +1472,79 @@ fn container_engine_logrotate(check: &mut Check) -> Result<CheckResult, failure:
     Ok(CheckResult::Ok)
 }
 
+fn storage_mounted_from_host(
+    check: &mut Check,
+    container_name: &'static str,
+    storage_directory_name: &'static str,
+) -> Result<CheckResult, failure::Error> {
+    lazy_static::lazy_static! {
+        static ref STORAGE_FOLDER_ENV_VAR_KEY_REGEX: Regex =
+            Regex::new("(?i)^storagefolder=(.*)")
+            .expect("This hard-coded regex is expected to be valid.");
+    }
+
+    let docker_host_arg = if let Some(docker_host_arg) = &check.docker_host_arg {
+        docker_host_arg
+    } else {
+        return Ok(CheckResult::Skipped);
+    };
+
+    let inspect_result = inspect_container(docker_host_arg, container_name)?;
+
+    let temp_dir = inspect_result
+        .config()
+        .and_then(docker::models::ContainerConfig::env)
+        .into_iter()
+        .flatten()
+        .find_map(|env| {
+            STORAGE_FOLDER_ENV_VAR_KEY_REGEX
+                .captures(env)
+                .and_then(|capture| capture.get(1))
+                .map(|match_| match_.as_str())
+        })
+        .unwrap_or(
+            // Hard-code the value here rather than using the tempfile crate. It needs to match .Net Core's implementation,
+            // and needs to be in the context of the container user instead of the host running `iotedge check`.
+            if cfg!(windows) {
+                r"C:\Windows\Temp"
+            } else {
+                "/tmp"
+            },
+        );
+
+    let storage_directory = Path::new(&*temp_dir).join(storage_directory_name);
+
+    let mounted_directories = inspect_result
+        .mounts()
+        .into_iter()
+        .flatten()
+        .filter_map(|mount| mount.destination().map(Path::new));
+
+    let volume_directories = inspect_result
+        .config()
+        .and_then(docker::models::ContainerConfig::volumes)
+        .map(std::collections::HashMap::keys)
+        .into_iter()
+        .flatten()
+        .map(Path::new);
+
+    if !mounted_directories
+        .chain(volume_directories)
+        .any(|container_directory| container_directory == storage_directory)
+    {
+        return Ok(CheckResult::Warning(
+            Context::new(format!(
+                "The {} module is not configured to persist its {} directory on the host filesystem.\n\
+                Data might be lost if the module is deleted or updated.",
```

Comment:
Should this give them a pointer on how to do it? Maybe a URL to our docs.

Reply:
Yeah, I was going to talk to Venkat about it when I saw him.